crucible-install.sh

```diff
@@ -58,6 +31,49 @@ EC_FAIL_AUTH_SET=13
 EC_FAIL_CHECKOUT=14
 EC_PUSHD_FAIL=15
 
+function determine_git_install_source {
+    local PWD SCRIPT_DIR
+    local git_status git_use_default git_tracking git_remote_branch git_remote_name git_remote_url
+
+    # check if the user specified GIT_REPO
+    if [ -n "${GIT_REPO}" ]; then
+        return
+    fi
+    git_use_default=0
+
+    PWD=$(pwd)
+    PWD=$(readlink -e ${PWD})
+    SCRIPT_DIR=$(dirname $0)
+    SCRIPT_DIR=$(readlink -e ${SCRIPT_DIR})
+    if [ "${PWD}" != "${SCRIPT_DIR}" ]; then
+        if ! pushd ${SCRIPT_DIR} > /dev/null; then
+            echo "WARNING: Failed to pushd to ${SCRIPT_DIR}"
+            git_use_default=1
+        fi
+    fi
+
+    git_status=$(git status --porcelain=2 --untracked-files=no --branch 2>&1)
+    if echo -e "${git_status}" | grep -q "not a git repository"; then
+        git_use_default=1
+    fi
+    if ! echo -e "${git_status}" | grep "branch\.upstream"; then
+        git_use_default=1
+    fi
+
+    if [ "${git_use_default}" == 0 ]; then
+        git_tracking=$(echo "${git_status}" | grep "branch\.upstream" | awk '{ print $3 }')
+        git_remote_branch=$(echo "${git_tracking}" | awk -F'/' '{ print $2 }')
+        git_remote_name=$(echo "${git_tracking}" | awk -F'/' '{ print $1 }')
+        git_remote_url=$(git remote get-url ${git_remote_name})
+
+        GIT_REPO="${git_remote_url}"
+        GIT_BRANCH="${git_remote_branch}"
```

Comment:
auto-detection case
you need to fork the main repo to test changes, there is no way to edit a file on the repo cloned with ssh and test it